src/Setup/WixBA/InstallationViewModel.cs

```diff
@@ -502,15 +452,21 @@ private void CacheAcquireComplete(object sender, CacheAcquireCompleteEventArgs e
 
         private void ExecutePackageBegin(object sender, ExecutePackageBeginEventArgs e)
         {
-            this.executePackageStart = e.ShouldExecute ? DateTime.Now : DateTime.MinValue;
+            lock (this)
```

Comment:
Never mind, not a big deal.  Each VM does not get its own thread and only one event handler will be running at a time, but better safe than sorry.